ports/rp2/machine_spi.c

```diff
@@ -231,7 +220,7 @@ STATIC void machine_spi_transfer(mp_obj_base_t *self_in, size_t len, const uint8
         channel_config_set_dreq(&c, spi_get_index(self->spi_inst) ? DREQ_SPI1_TX : DREQ_SPI0_TX);
         dma_channel_configure(chan_tx, &c,
             &spi_get_hw(self->spi_inst)->dr,
-            src,
+            write_only ? src : &dev_null,
```

Comment:
why is this needed?

Reply:
hmm... that's something that came in with https://github.com/micropython/micropython/pull/8021 / external version I initially based this PR on. I can't see a need for it myself, I'll remove and re-test.